src/main/java/io/cryostat/net/web/http/api/beta/RecordingUploadPostHandler.java

```diff
@@ -136,6 +143,22 @@ public boolean isAsync() {
         return false;
     }
 
+    @Override
+    public SecurityContext securityContext(RequestParameters params) {
+        String sourceTarget = params.getPathParams().get("sourceTarget");
+        String recordingName = params.getPathParams().get("recordingName");
+        try {
+            return recordingArchiveHelper.getRecordings(sourceTarget).get().stream()
+                    .filter(r -> r.getName().equals(recordingName))
```

Comment:
I think from the number of get SecurityContext methods from Metadata from a recordingstream can be refactored into a new method in `RecordingArchiveHelper`?
like:

```java
SecurityContext getSecurityContext(List<ArchivedRecordingInfo> recordings, String recordingName) {
              return recordings.stream()
                    .filter(r -> r.getName().equals(recordingName))
                    .findFirst()
                    .map(ArchivedRecordingInfo::getMetadata)
                    .map(Metadata::getSecurityContext)
                    .orElseThrow(() -> new ApiException(404));
}
``` 

and then call them like 
```java
List<ArchivedRecordingInfo> recordings = rah.getRecordings(sourceTarget).get();
return recordingArchiveHelper.getSecurityContext(recordings, recordingName);
```